gym_snake/envs/snake_env.py

```diff
@@ -30,24 +30,21 @@ def step(self, action: spaces.Discrete(4)) -> tuple:
         err_msg = "%r (%s) invalid" % (action, type(action))
         assert self.action_space.contains(action), err_msg
 
-        # Get current Observation
-        observation = self.game.get_board()
-
+        # Play one frame of Snake Game
+        # If there was a fruit collision during last frame, move the fruit.
+        if self.game.check_fruit_collision():
+            self.game.respond_to_fruit_consumption() #FIXME: @jackdavidweber pick up here
         self.game.clock.tick(self.game.fps)
-        
         self.game.move_snake(action)
-        rewards = self.game.check_collisions()
+    
+        # Get observation after move
+        observation = self.game.get_board()
 
-        if rewards == 1:
-            self.game.respond_to_fruit_consumption() #FIXME: @jackdavidweber pick up here
-        
-        done = self.game.restart
+        # Get rewards
+        rewards = self.game.check_collisions()
 
-        if self.game.restart == True:
-            #  FIXME: restart is used to indicate whether game was forced over during
-            # self.game.check_collisions(). This seems to be a non-ideal way of achieving this.
-            self.game.restart = False
-            # continue FIXME: this was within a loop originally
+        # Game is over if wall collision or body collision occurred. TODO: add end done for time limit
+        done = self.game.check_wall_collision() or self.game.check_body_collision()
```

Comment:
`done = rewards == -1` is equivalent to `done = self.game.check_wall_collision() or self.game.check_body_collision()` without running the actual logic again since it is already run once in `rewards = self.game.check_collisions()` on line 44. 

The way it is currently written is a bit more clear, but just wanted to point out the duplicated logic. Your call on whether style vs. non-duped logic is best here!

Reply:
I think it just is a lot more readable this way which is why I have it. But appreciate the comment